descwl_coadd/tests/test_coadd.py

```diff
@@ -166,6 +172,61 @@ def test_coadds_with_interpolator_smoke(buff):
         assert np.all(coadd_data['coadd_mfrac_exp'].image.array == 0)
 
 
+@pytest.mark.parametrize('buff', [4, 5])
+def test_coadd_equality_with_dm_interpolator(buff):
+    """Test that the DM-implementation of the interpolator behaves identical.
+    """
+    rng = np.random.RandomState(123)
+
+    coadd_dim = 101
+    psf_dim = 51
+
+    bands = ['r', 'i', 'z']
+    sim_data = _make_sim(
+        rng=rng, psf_type='gauss', bands=bands,
+        coadd_dim=coadd_dim, psf_dim=psf_dim,
+    )
+
+    our_interpolator = CTInterpolator(buff=buff)
+
+    config = CloughTocher2DInterpolateTask.ConfigClass()
+    config.interpLength = buff
+    config.badMaskPlanes = FLAGS2INTERP
+
+    dm_interpolator = CloughTocher2DInterpolateTask(config=config)
+
+    # coadd each band separately
+    bdata = sim_data['band_data']
+    for band in bands:
+        assert band in bdata
+        exps = bdata[band]
+
+        coadd_data = make_coadd(
+            exps=exps,
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=our_interpolator,
+        )
+
+        dm_coadd_data = make_coadd(
+            exps=exps,
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+            interpolator=dm_interpolator,
+        )
+
+        # Compare only the main exposure since the noise realizations are
+        # different and mfrac is not affected by the interpolator.
+        assert (coadd_data['coadd_exp'].image.array ==
+                dm_coadd_data['coadd_exp'].image.array).all()
```

Comment:
This level of agreement is surprising in that you'd need the set of bad pixels around the good ones to match exactly. 

Let's add an assert that some data was actually interpolated in the test. Otherwise things would appear to agree but it'd be because we didn't do any interpolation.

Reply:
The logic of selecting the bad pixels is exactly the same and involves only integer tuples. The floating point operations are all done by the `scipy` function, so I am not surprised that the agreement is exact. But I'll add a test to see that some pixels are actually interpolated.

Reply:
Yes, I said that we have a different implementation of this function in the DM codebase: https://github.com/LSSTDESC/descwl_coadd/blob/master/descwl_coadd/interp.py#L13-L14

Once we get the location of good pixels and bad pixels, we call the `scipy.CloughTocher2DInterpolator` the same way as done here.

Reply:
FWIW, here's the C++ implementation of the function that behaves identical to the function I posted above: https://github.com/lsst/meas_algorithms/blob/main/src/CloughTocher2DInterpolatorUtils.cc

Reply:
Good catch with the pixels not being interpolated in the first place, Matt. I'm going to do some more local testing and raise it again later. Marking it as a draft for now.